storage/storage_tests/storage_tests.cpp

```diff
@@ -1069,4 +1070,56 @@ UNIT_TEST(StorageTest_IsPointCoveredByDownloadedMaps)
     TEST(IsPointCoveredByDownloadedMaps(montevideoUruguay, storage, *countryInfoGetter), ());
   }
 }
+
+UNIT_TEST(StorageTest_TwoInstance)
+{
+  Platform & platform = GetPlatform();
+  string const writableDir = platform.WritableDir();
+
+  string const testFolder1 = string("testfolder1");
+  Storage storage1(COUNTRIES_FILE, testFolder1);
```

Comment:
Why not to create subclass of Storage (like, TestStorage), which will call Platform::SetWritableDirForTesting() ? After that you'll be able to get rid of this explicit storage specification?